src/components/SearchBar.tsx

```diff
@@ -51,7 +51,27 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
             let data = new CategoryData(c, "CategoryCheckbox", true, this.onCategoriesChanged.bind(this));
             data.onOnlyButtonClicked = this.onOnlyButtonClicked.bind(this);
             this.categoryData.push(data);
-        }.bind(this))
+        }.bind(this));
+    }
+
+    //To receive new prop with updated categories
+    componentWillReceiveProps(newProps: SearchBarProps) {
+        let oldState = this.state;
+        this.setState({ selectedCategories: newProps.categories });
+
+        let oldCategoryData = this.categoryData;
+        this.categoryData = [];
+        _.each(newProps.categories, function (c: string) {
+            let oldCategory = oldCategoryData.find((x) => {
+                return x.name === c;
+            });
+            let data = new CategoryData(c, "CategoryCheckbox", true, this.onCategoriesChanged.bind(this));
+            data.onOnlyButtonClicked = this.onOnlyButtonClicked.bind(this);
+            if (oldCategory) {
+                data.checked = oldCategory.checked;
+            }
+            this.categoryData.push(data);
```

Comment:
I'd encourage to place `let oldCategory = ...` right before `if (oldCategory) ...` because now they look disconnected.